src/dotnet/APIView/APIViewWeb/Filters/ApiKeyAuthorizeAsyncFilter.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace APIViewWeb.Filters
+{
+    public class ApiKeyAuthorizeAsyncFilter : Attribute, IAsyncAuthorizationFilter
+    {
+        private static string _apiKeyHeader = "ApiKey";
+        private string _azure_sdk_bot = "azure-sdk";
+        private string _apiKeyValue;
+
+        public ApiKeyAuthorizeAsyncFilter(IConfiguration configuration)
+        {
+            var apiKey = configuration[_apiKeyHeader];
+            if (!string.IsNullOrEmpty(apiKey))
+            {
+                _apiKeyValue = apiKey;
+            }
+        }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        public async Task OnAuthorizationAsync(AuthorizationFilterContext context)
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+        {
+            var request = context.HttpContext.Request;
+            var hasApiKeyHeader = request.Headers.TryGetValue(_apiKeyHeader, out var apiKeyValue);
+            if (hasApiKeyHeader && apiKeyValue == _apiKeyValue)
+            {
+                var apiKeyClaim = new Claim("apikey", apiKeyValue);
```

Comment:
do we need to attach the key as a claim?

Reply:
Removed.